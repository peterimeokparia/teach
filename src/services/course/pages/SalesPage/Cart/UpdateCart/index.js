import { 
connect } from 'react-redux';

import { 
getCoursesByCourseIdSelector  } from 'services/course/selectors';

import useSalesHook from 'services/course/pages/SalesPage/hooks/useSalesHook';

const UpdateCart = ({ course }) => {  
    let {
        removeFromCart,
        returnToCourseListPage,
    } = useSalesHook({ course });

return ( <div> 
            <div> {`are you sure you want to remove ${course?.name} from your cart?`} </div> 
                <div> 
                    <button onClick={removeFromCart}>Yes</button>
                    <button onClick={returnToCourseListPage}>No</button>
                </div>
            </div> 
        );
};

const mapState = (state, props)  => ({
    currentUser : state.users.user,
    course: getCoursesByCourseIdSelector(state, props),
});

export default connect(mapState, null)( UpdateCart );