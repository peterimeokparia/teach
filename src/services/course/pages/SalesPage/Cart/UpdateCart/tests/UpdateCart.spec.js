import React from 'react';

import { 
Provider, 
useSelector } from "react-redux";

import { 
applyMiddleware } from 'redux'; 

import { 
mockStoreObject } from 'Services/course/api';

import UpdateCart  from 'services/course/pages/SalesPage/Cart/UpdateCart';
import renderer from 'react-test-renderer';
import thunk from "redux-thunk"; 
import * as reactRedux from 'react-redux';
import configureStore from 'redux-mock-store';

  const mockStore = configureStore([]);
  
describe('Updates a shopping cart.', () =>  {  

  let  store = mockStore(mockStoreObject, applyMiddleware(thunk));

  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');
 
  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
    jest.spyOn(Date.prototype, 'toLocaleTimeString')
      .mockImplementation(() => '12:00:00');
    jest.spyOn(Date.prototype, 'toLocaleDateString')
      .mockImplementation(() => 'AM 1/01/2021');
  });

  it('renders as expected', async () => {

    const dispatch = jest.fn();
    useDispatchMock.mockReturnValue(dispatch)

    const tree = renderer.create(
      <Provider store={store}>
        <UpdateCart /> 
    </Provider>
    );
      expect(tree.toJSON()).toMatchSnapshot();
  })
});

