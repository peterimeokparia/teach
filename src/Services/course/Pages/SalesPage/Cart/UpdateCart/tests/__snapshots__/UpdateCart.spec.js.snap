// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Updates a shopping cart. renders as expected 1`] = `
<div>
  <div>
     
    are you sure you want to remove undefined from your cart?
     
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Yes
    </button>
    <button
      onClick={[Function]}
    >
      No
    </button>
  </div>
</div>
`;
